gaphor/UML/treemodel.py

```diff
@@ -139,6 +156,26 @@ def __iter__(self):
         yield from self.elements
         yield from self.relationships
 
+    def on_format_text_changed(self, tree_item: TreeItem):
+        if (
+            isinstance(tree_item.element, UML.Parameter)
+            and tree_item.element.activityParameterNode
+        ):
+            for (
+                activity_parameter_node
+            ) in tree_item.element.activityParameterNode.items:
+                found_tree_item = self.get_tree_item(activity_parameter_node)
+                if found_tree_item is not None:
+                    found_tree_item.sync()
+
+    def get_tree_item(self, element: Base) -> TreeItem | None:
+        tree_item = next((ti for ti in self.elements if ti.element is element), None)
+        if tree_item is None:
+            tree_item = next(
+                (ti for ti in self.relationships if ti.element is element), None
+            )
+        return tree_item
```

Comment:
I'm not sure if this will really work: it's only looking up elements in the current branch. But `Parameter` is owned by an `ActivityParameterNode`, so it should look in the parent branch.

The idea is pretty good though. We need something similar for attributes, operations, and a couple of other types that format based on owned and related elements.

Maybe we should change `TreeModel.on_attribute_changed` to also sync all owner types?